subsys/mgmt/mcumgr/lib/cmd/fs_mgmt/src/fs_mgmt.c

```diff
@@ -413,22 +421,26 @@ fs_mgmt_file_hash_checksum(struct mgmt_ctxt *ctxt)
 
 		if (group->byte_string == true) {
 			/* Output is a byte string */
-			ok &= zcbor_bstr_encode_ptr(zse, output, group->output_size);
+			ok &= zcbor_bstr_encode_ptr(zse, output.str, group->output_size);
 		} else {
 			/* Output is a number */
 			uint64_t tmp_val = 0;
 
 			if (group->output_size == sizeof(uint8_t)) {
-				tmp_val = (uint64_t)(*(uint8_t *)output);
+				tmp_val = (uint64_t)output.u8;
 			} else if (group->output_size == sizeof(uint16_t)) {
-				tmp_val = (uint64_t)(*(uint16_t *)output);
+				tmp_val = (uint64_t)output.u16;
 			} else if (group->output_size == sizeof(uint32_t)) {
-				tmp_val = (uint64_t)(*(uint32_t *)output);
+				tmp_val = (uint64_t)output.u32;
+#if FS_MGMT_CHECKSUM_HASH_LARGEST_OUTPUT_SIZE >= 8
```

Comment:
This check is invalid, the output size does not have any relation to this, it would need a new numerical define adding which contains the largest numerical checksum, the current define is combined for string and numerical data, so this code would always be enabled for sha256 despite the largest numerical method being crc32, which is 4 bytes